### PR TITLE
Add columnar format diagrams and introduction on row vs column

### PR DESCRIPTION
**Pull request by raulcd**

### CODE REVIEW

File: columnar_format.ipynb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    "\n",
+    "That means that you have all the information for every row together in memory. This is great for transactional Transactional Databases ([OLTP](https://en.wikipedia.org/wiki/Online_transaction_processing)) and if you want to access all the data information for a row every time.\n",
+    "\n",
+    "### Columnar format\n",
+    "\n",
+    "If we have a much bigger table and we just want, for example the average cost of transaction skipping all the data that is irrelevant to do that computation would be costly. That's why storage and memory representation for Columnar format is important.\n",
+    "\n",
+    "![image info](./diagrams/column_format.svg)\n",
+    "\n",
+    "For Analytical Processing Databases ([OLAP](https://en.wikipedia.org/wiki/Online_analytical_processing)) we use columnar format to easily perform computations and analysis over the data types.\n",
```
Comment (jorisvandenbossche) [inline columnar_format.ipynb:32]: Do we know if this is always true? I am not super familiar with how and when the OLAP term is used (but from the wiki page, it also seems it's an old term). 
We could also say something a bit less generic like "Modern analytical processing databases (OLAP) typically use a columnar format ..."

File: columnar_format.ipynb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+   "cell_type": "markdown",
+   "id": "e783b4f6-a25f-44b9-8e1f-122759b8d349",
+   "metadata": {},
+   "source": [
+    "# Arrow Columnar Format\n",
+    "\n",
+    "\n",
+    "## Why does Arrow use a Columnar in-memory format?\n",
+    "\n",
+    "When we discuss about Row vs Columnar format we usually discuss about the storage data type. With Arrow we talk about the data structures in memory directly.\n",
```
Comment (jorisvandenbossche) [inline columnar_format.ipynb:13]: What do you exactly mean here with "storage data type"? 
"data type" already has a specific meaning in context of Arrow as well, but I suppose that is not meant here, which makes it a bit confusing. Something like "... about the way the data is stored"?

Comment (AlenkaF) [reply columnar_format.ipynb:13]: I would remove two occurences of "about" in the first sentence.

**Comment by jorisvandenbossche** [comment]
We can address the comments in a follow-up

**Pull request merged**